tests/Teams/ProactiveMessages/Bots/ProactiveMessageBot.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.BotBuilderSamples.Bots
+{
+    public class ProactiveMessageBot : TeamsActivityHandler
+    {
+        private string _appId;
+
+        public ProactiveMessageBot(IConfiguration configuration)
+        {
+            _appId = configuration["MicrosoftAppId"];
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            // You can hand roll a connector to manually address a proactive message
+            var connectorClient = turnContext.TurnState.Get<IConnectorClient>();
+            var connector = new ConnectorClient(connectorClient.Credentials);
+
+            connector.BaseUri = new Uri(turnContext.Activity.ServiceUrl);
+
+            var parameters = new ConversationParameters
+            {
+                Bot = turnContext.Activity.From,
+                Members = new ChannelAccount[] { turnContext.Activity.From },
+                ChannelData = new TeamsChannelData
+                {
+                    Tenant = new TenantInfo
+                    {
+                        Id = turnContext.Activity.Conversation.TenantId,
+                    },
+                },
+            };
+
+            var converationReference = await connector.Conversations.CreateConversationAsync(parameters);
+            var proactiveMessage = MessageFactory.Text($"Hello {turnContext.Activity.From.Name}. You sent me a message. This is a proactive responsive message.");
+            proactiveMessage.From = turnContext.Activity.From;
+            proactiveMessage.Conversation = new ConversationAccount
+            {
+                Id = converationReference.Id.ToString(),
+            };
+
+            await connector.Conversations.SendToConversationAsync(proactiveMessage, cancellationToken);
+
+            // Or you can use the adaptor to send a message if you already have a conversation reference. You can put this code into the controller if
```

Comment:
Might as well change `"adaptor"` to be spelled `"adapter"` since the member property below is `Adapter`

Reply:
Updated